compiler/modulegraphs.nim

```diff
@@ -343,31 +364,44 @@ proc hash*(u: SigHash): Hash =
 
 proc hash*(x: FileIndex): Hash {.borrow.}
 
+template getC(): untyped =
+  when c is PContext: c
+  else: c.c
```

Comment:
Too ugly... Can we find a better solution?

Reply:
this is pre-existing and I've strictly improved things in this PR via `getC` by avoiding this:
```
    when compiles(c.c.graph):
      if c.c.graph.onUsage != nil: c.c.graph.onUsage(c.c.graph, s, info)
    else:
      if c.graph.onUsage != nil: c.graph.onUsage(c.graph, s, info)
```
which had `compiles` and duplicates code.

But I've added a future work item to further improve this.

Reply:
It was only code for "nimfind" which is dead code now that we got "nim check --defusages"

Reply:
@araq i see you've removed nimfind in https://github.com/nim-lang/Nim/pull/17737, but please don't remove `onDef` + friends, they're very useful during debugging (and i wrote a competitor tool of nimfind that added more features, which relied on those onDef, onUsage etc.

Reply:
I prefer to distinguish between `nkSym` and `nkSymDef` instead but it would break the macro system until we have an AST to AST translation step. :-/